DuckDuckGo/NetworkProtectionAccessController.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  NetworkProtectionAccessController.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#if NETWORK_PROTECTION
+
+import Foundation
+import BrowserServicesKit
+import ContentBlocking
+import Core
+import NetworkProtection
+import Waitlist
+
+enum NetworkProtectionAccessType {
+    /// Used if the user does not have waitlist feature flag access
+    case none
+
+    /// Used if the user has waitlist feature flag access, but has not joined the waitlist
+    case waitlistAvailable
+
+    /// Used if the user has waitlist feature flag access, and has joined the waitlist
+    case waitlistJoined
+
+    /// Used if the user has been invited via the waitlist, but needs to accept the Privacy Policy and Terms of Service
+    case waitlistInvitedPendingTermsAcceptance
+
+    /// Used if the user has been invited via the waitlist and has accepted the Privacy Policy and Terms of Service
+    case waitlistInvited
+
+    /// Used if the user has been invited to test Network Protection directly
+    case inviteCodeInvited
+}
+
+protocol NetworkProtectionAccess {
+    func networkProtectionAccessType() -> NetworkProtectionAccessType
+}
+
+struct NetworkProtectionAccessController: NetworkProtectionAccess {
```

Comment:
I've extracted the logic that determines whether NetP is visible, and the logic to disable it once the test is over.